apps/e2e/integration/reconciliation_order.spec.ts

```diff
@@ -1,90 +1,243 @@
+		await isbnInput.fill(placedOrderLines[0].isbn);
+		await page.keyboard.press("Enter");
+
+		// Move to comparison view
+		await page.getByRole("button", { name: "Compare" }).click();
+
+		// Verify comparison table structure
+		const table = page.getByRole("table");
+
+		// Check supplier name row
+		const supplierNameRow = table.getByRole("row").nth(1);
+		await expect(supplierNameRow.getByRole("cell", { name: placedOrderLines[0].supplier_name })).toBeVisible();
+
+		// Check book details row
+		const bookRow = table.getByRole("row").nth(2);
+		await expect(bookRow.getByRole("cell", { name: placedOrderLines[0].isbn })).toBeVisible();
+
+		// Verify delivery stats show under-delivery
+		await expect(page.getByText("Total delivered:")).toBeVisible();
+		await expect(page.getByText("1 / 2")).toBeVisible(); // Assuming 2 were ordered, 1 scanned
+
+		// Optional: Verify any visual indicators of under-delivery
+		// This depends on your UI implementation
+		// await expect(bookRow.getByRole("cell").nth(4)).toHaveClass("text-warning"); // Example
+	}
+);
+testOrders("should show unmatched deliveriesy when ordered books do not match scanned books", async ({ page, placedOrderLines, books }) => {
+	// Navigate to reconciliation
+	await page.goto(`${baseURL}orders/suppliers/orders/`);
+
+	await page.getByText("Ordered").nth(1).click();
+	await page.getByRole("checkbox").nth(1).click();
+
+	await page.getByText("Reconcile").first().click();
+
+	const isbnInput = page.getByPlaceholder("Enter ISBN of delivered books");
+
+	// Scan non ordered books
+	await isbnInput.fill(placedOrderLines[0].isbn);
+	await page.keyboard.press("Enter");
+
+	const table = page.getByRole("table");
+	await expect(table.getByText(placedOrderLines[0].isbn)).toBeVisible();
+
+	await isbnInput.fill(books[2].isbn);
+	await page.keyboard.press("Enter");
+
+	await expect(table.getByText(books[2].isbn)).toBeVisible();
+
+	// Move to comparison view
+	await page.getByRole("button", { name: "Compare" }).nth(1).click();
+
+	// Verify comparison table structure
+
+	// Check supplier name row
+	const unmatchedRow = table.getByRole("row").nth(1);
+	await expect(unmatchedRow.getByRole("cell", { name: "unmatched" })).toBeVisible();
+
+	// Check book details row
+	const unmatchedBookRow = table.getByRole("row").nth(2);
+	await expect(unmatchedBookRow.getByRole("cell", { name: books[2].isbn })).toBeVisible();
+
+	const supplierNameRow = table.getByRole("row").nth(3);
+	await expect(supplierNameRow.getByRole("cell", { name: placedOrderLines[0].supplier_name })).toBeVisible();
+
+	const matchedBookRow = table.getByRole("row").nth(4);
+	await expect(matchedBookRow.getByRole("cell", { name: placedOrderLines[0].isbn })).toBeVisible();
+
+	await expect(page.getByText("Total delivered:")).toBeVisible();
 	await expect(page.getByText("2 / 1")).toBeVisible();
 });
+testOrders("should show correct delivery stats in commit view", async ({ page, books, placedOrderLines }) => {
```

Comment:
Doesn't seem to get to the "commit" stage - should it not be testing what shows up in the modal? It uses a random isbn and `getByText` for table/row assertions.

Additionally, we should add assertions for what we expect to happen after committing. I know this is WIP, but just add what you can and leave a todo note